fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/HashJobStore.java

```diff
@@ -51,4 +62,58 @@ public HashJobStore() {
     public Observable<JobRow> getAllRows() {
         return Observable.from(store.values());
     }
+
+    @Override
+    public Observable<JobRow> getFilteredRows(Set<JobRowFilter> jobRowFilters)
+            throws IllegalArgumentException {
+        return getAllRows().filter(jobRow -> satisfiesFilters(jobRowFilters, jobRow));
+    }
+
+    /**
+     * This method checks if the given JobRow satisfies all the JobRowFilters and returns true if it does.
+     * If a JobField in any of the filters is not a part the JobRow, this method throws a IllegalArgumentException.
+     *
+     * @param jobRowFilters  A Set of JobRowFilters specifying the different conditions to be satisfied
+     * @param jobRow  The JobRow which needs to be inspected
+     *
+     * @return true if the JobRow satisfies all the filters, false otherwise
+     *
+     * @throws IllegalArgumentException if a JobField in any of the filters is not a part the JobRow
+     */
+    private boolean satisfiesFilters(Set<JobRowFilter> jobRowFilters, JobRow jobRow)
+            throws IllegalArgumentException {
+        return jobRowFilters.stream().allMatch(filter -> satisfiesFilter(jobRow, filter));
+    }
+
+    /**
+     * This method checks if the given JobRow satisfies the given JobRowFilter and returnd true if it does.
+     * If a JobField in the filter is not a part the JobRow, this method throws a IllegalArgumentException.
+     *
+     * @param jobRow  The JobRow which needs to be inspected
+     * @param jobRowFilter  A JobRowFilter specifying the filter condition
+     *
+     * @return true if the JobRow satisfies the filter, false otherwise
+     *
+     * @throws IllegalArgumentException if a JobField in the filter is not a part the JobRow
+     */
+    private boolean satisfiesFilter(JobRow jobRow, JobRowFilter jobRowFilter) throws IllegalArgumentException {
+        JobField filterJobField = jobRowFilter.getJobField();
+        FilterOperation filterOperation = jobRowFilter.getOperation();
+        if (!filterOperation.equals(FilterOperation.eq)) {
+            LOG.debug(FILTER_OPERATOR_INVALID.logFormat(filterOperation));
+            throw new IllegalArgumentException(FILTER_OPERATOR_INVALID.format(filterOperation));
+        }
+        if (!jobRow.containsKey(filterJobField)) {
+            Set<JobField> actualJobFields = jobRow.keySet();
+            LOG.debug(JOBFIELD_NOT_PRESENT_IN_JOB_META_DATA.logFormat(filterJobField, actualJobFields));
+            throw new IllegalArgumentException(
+                    JOBFIELD_NOT_PRESENT_IN_JOB_META_DATA.format(filterJobField, actualJobFields)
+            );
```

Comment:
The `HashJobStore` should probably be supporting every filter operation. It shouldn't be too difficult, because the values of the `JobRow` map are all Strings, so the filter operations all translate directly into String operations.